jib-core/src/main/java/com/google/cloud/tools/jib/frontend/BuildStepsRunner.java

```diff
@@ -37,21 +39,49 @@
 import java.util.function.Supplier;
 import org.apache.http.conn.HttpHostConnectException;
 
-/** Runs {@link BuildImageSteps} and builds helpful error messages. */
-public class BuildImageStepsRunner {
+/** Runs a {@link BuildSteps} and builds helpful error messages. */
+public class BuildStepsRunner {
+
+  private final Supplier<BuildSteps> buildStepsSupplier;
+
+  /**
+   * Creates a runner to build an image. Creates a directory for the cache, if needed.
+   *
+   * @param useOnlyProjectCache if {@code true}, sets the base layers cache directory to be the same
+   *     as the application layers cache directory
+   * @throws CacheDirectoryCreationException if the {@code cacheDirectory} could not be created
+   */
+  public static BuildStepsRunner forBuildImage(
+      BuildConfiguration buildConfiguration,
+      SourceFilesConfiguration sourceFilesConfiguration,
+      Path cacheDirectory,
+      boolean useOnlyProjectCache)
+      throws CacheDirectoryCreationException {
+    Initializer cacheInitializer = getCacheInitializer(cacheDirectory, useOnlyProjectCache);
+    return new BuildStepsRunner(
+        () -> new BuildImageSteps(buildConfiguration, sourceFilesConfiguration, cacheInitializer));
```

Comment:
If this is just passing in a new `BuildImageSteps`, it doesn't seem like we need the supplier.